ipaserver/plugins/cert.py

```diff
@@ -787,6 +800,62 @@ def execute(self, csr, all=False, raw=False, **kw):
         )
 
 
+def _dns_name_matches_principal(name, principal, principal_obj):
+    """
+    Ensure that a DNS name matches the given principal.
+
+    :param name: The DNS name to match
+    :param principal: The subject ``Principal``
+    :param principal_obj: The subject principal's LDAP object
+    :return: True if name matches, otherwise False
+
+    """
+    for alias in _collect_principal_aliases(principal_obj):
+        # we can only compare them if both subject principal and
+        # the alias are service or host principals
+        if not (alias.is_service and principal.is_service):
+            continue
+
+        # ignore aliases with different realm or service name from
+        # subject principal
+        if alias.realm != principal.realm:
```

Comment:
I'm going to leave it in anyway.  It does no harm and is one less potential bug if we do ever support aliases with different realms, somehow.